### ISSUE TITLE
Automate release initiation (checklist, drafts, etc.)

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#14003]
Related to #13610.

## Description
Currently we're not very consistent about initiating the release and posting the [checklist](https://github.com/ethereum/solidity/blob/develop/ReleaseChecklist.md). Sometimes it goes up late, sometimes earlier. We should have a one-button solution for the release captains initiate the release preparations.

## Specification
- [ ] Create a github action that posts the content of the [checklist](https://github.com/ethereum/solidity/blob/develop/ReleaseChecklist.md) in a new issue.
    - The title should include the version number of the upcoming release. For simplicity assume it's always a patch release, we'll update manually if it's not.
- [ ] The action should be triggered via [`workflow_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch), so that someone with write access to the repo can click the button there.
    - Make sure it's only available to people with write access. If it's available to anyone, we'll need a check in the action to prevent that.
    - (Nice to have) Include a check that a checklist issue is not already open to prevent accidental double posting.
- [ ] If necessary, adjust the checklist template to be suitable for posting in an issue. The way it is now, it was done with an assumption that we do some manual adjustments before copying it (e.g. removing the prerequisites).

## Extras
The scope is intentionally limited here but in the future we may want to have the action do more things. E.g. create a draft of the release. Maybe even create a draft of a blog post. That is yet to be determined, for now we want to start with something simple and quick to implement.

Another idea might be to post separate issues for pre-flight steps and for the actual checklist to ensure that they're not done in parallel.
